CONTRIBUTING.md

```diff
@@ -1,87 +1,124 @@
+**New implementations** are welcome! This includes: new solutions for a problem, different representations for a data structure, algorithm design with a different complexity or features.
 
-Algorithms should:
+**Improving documentation and comments** and **adding tests** is also highly welcome.
 
-* have intuitive object and function names that make their purpose clear to readers
-* use Nim naming conventions and intuitive variable names with correct typing to ease comprehension
-* Don't use unsigned numerical types (`uint` and its sized variants), unless wrapping behaviour or binary manipulation is required for the algorithm.
-* raise Nim exceptions (`ValueError`, etc.) on erroneous input values
-* The side effects must be annotated
-* have documentation comments with clear explanations and/or URLs to source materials
-* contain tests (runnableExamples, std/unittest) that test both valid and erroneous input values
-* return all calculation results instead of printing or plotting them
+**Identical implementations** are not allowed.
 
-Please focus hard on the naming of functions, objects, and variables.  Help your reader by using **descriptive names** that can help you to remove redundant comments.
-  - Single letter variable names must be avoided unless your variable has a short lifespan. If your variable come from a mathematical context, no confusion is possible with another variable, you may use single-letter variables. When there are multiple indices (matrix, sums integrals) you may use single-letter variables. For string algorithms, `c` for character is a common use. Pay attention though not creating any confusion when dealing with multiple characters.
-  You can use `index` instead of `i` when iterating in a `for` loop.
-  - For some parameters, using short variable names enable to distinguish the inputs easily from the parameters.
-  - Expand acronyms. Prefer `greatestCommonDivisor()` to `gcd()`, as the former is easier to understand than the latter, especially for non-native English speakers.
-  - The variable and function names should be camelCase while object names should be PascalCase.
-- Pay attention to spacing around operators.
-- Avoid importing external libraries for basic algorithms. Only use those libraries for complicated algorithms.
+### Implementation requirements
 
-Algorithms in this repo should not be how-to examples for existing Nim packages. Instead, they should perform internal calculations or manipulations to convert input values into different output values.  Those calculations or manipulations can use objects or functions of existing Nim modules but each algorithm in this repo should add unique value.
+- The unit of an implementation we expect is a **Nim module**, with all the corresponding consequences. Although the main goals of this repository are educational, the module form mirrors a real world scenario and makes it easy to use the code from this repository in other projects.
```

Comment:
I would remove :
```suggestion
- The unit of an implementation we expect is a **Nim module**. Although the main goals of this repository are educational, the module form mirrors a real world scenario and makes it easy to use the code from this repository in other projects.
```
which is too vague, or link to the official documentation:
```suggestion
- The unit of an implementation we expect is a **Nim module**, [with all the corresponding consequences](https://nim-lang.org/docs/manual.html#modules). Although the main goals of this repository are educational, the module form mirrors a real world scenario and makes it easy to use the code from this repository in other projects.
```

Reply:
Done.